src/components/views/read/cvForm/types.ts

```diff
@@ -53,6 +53,7 @@ export type PortfolioObj = {
 
 export type MotivationObj = {
   motivationLetter: string;
+  saved: boolean;
```

Comment:
Storing the value of saved would let me know as this prop won't be changed since it's stored inside an appData so it won't get re-defined each time we are at that section, to toggle the user is done from editing or not.

Reply:
And showFields state is defined within the section, so it's local, thus just being used to handle showing the fields at a certain moment within the section.

Reply:
I am not sure what exact problem it solves.

Reply:
So, showFields is working as local state which will be defined each time the user goes on that section, to handle only showing the fields of the cards, based on the user actions of editing and also validation, when there is some issues with the fields the user missed to fill up. And saved prop is to handle the state of whenever the user is done from editing, it will be stored in the appData object as true "saved is fine".

Reply:
If the user has to click "Done" on a card (changes the `showFields` value for that card) to save the data, if all your showFields are set to false, you can be sure that you have saved all the data. 
I think the use of `saved` key is redundant.

Reply:
My main issue that you never turn the `saved` key to `false`, so from my point of view it is useless.

Reply:
saved prop is supposed to be initally false, and turned to true when everything is valid and good to go, also showFields might not be a good handle for the saving state, since it would be initialized to false, and if both logics used together, it might get overlaped. that's my opinion

Reply:
Yes I agree for that saved is initially false. But once it is set to true, you  do not set it to false if you have unsaved data, since you do not save the appData if it is not valid. Do you see ? This is why for me it does not make sens to store in the appData whether the appData is "valid of not". Because by essence the appData should always be valid. only on the client-side inside the state (which is volatile) could an object (that would eventually become and appData if posted) be "invalid". So it makes sense for me to use a state that tracks whether things are saved or not. So I would say, do not mix, simply remove the `saved` and use `showFileds` to know what has been saved. This also means that you can't create "dummy" appData on step transition (will need to be changed).